### PR TITLE
0.5.0/events

### PR DESCRIPTION
**Pull request by med-amine-bj**

### CODE REVIEW

File: Usiqee/Utils/Ressources/Translation/L10N.swift [hunk 1]
```diff
@@ -20,6 +20,18 @@ struct L10N {
...
+            static let ep: String = "ep"
+            static let album: String = "album"
+            static let single: String = "single"
+            static let mixtape: String = "mixtape"
+            static let freestyle: String = "freestyle"
+            static let video: String = "video"
+            static let concert: String = "concert"
+            static let showcase: String = "showcase"
+            static let festival: String = "festival"
+            static let other: String = "other"
```
Comment (guaire94) [inline Usiqee/Utils/Ressources/Translation/L10N.swift:33]: can we localize them with this format EVENTS_{{TYPE}} ?

File: Usiqee/Manager/ManagerEvents.swift [hunk 2]
```diff
@@ -0,0 +1,156 @@
+//
+//  ManagerEvents.swift
+//  Usiqee
+//
+//  Created by Amine on 27/05/2021.
+//
+
+import Foundation
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Manager/ManagerEvents.swift:8]: yeah, needed for `Date`

File: Usiqee/Module/Agenda/EventsFilter/EventsFilterVC.swift [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch tableviewHandler.sectiontype(at: section) {
+        case .followed:
+            return 1
+        case .event:
+            return tableviewHandler.events.count
+        default:
+            return 0
+        }
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Module/Agenda/EventsFilter/EventsFilterVC.swift:88]: can't be done directly inside `SectionType` cause event isnt seen from inside the enum but I added as a helper in the tableview helper

File: Usiqee/Module/Agenda/EventsFilter/EventsFilterVC.swift [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        guard let cellType = tableviewHandler.item(for: indexPath) else {
+            return 0
+        }
+        
+        switch cellType {
+        case .followed:
+            return EventsFilterSwitchCell.Constants.height
+        case .event:
+            return EventsFilterEventTypeCell.Constants.height
+        }
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Module/Agenda/EventsFilter/EventsFilterVC.swift:127]: same

File: Usiqee/Module/Global/Views/MonthYearDatePickerView.swift [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+class MonthYearDatePickerView: UIPickerView {
+    
+    // MARK: - Enums
+    private enum DatePickerComponent : Int {
+        case month, year
+    }
+    
+    private enum Constants {
+        fileprivate static let months = ["Janvier", "Février", "Mars", "Avril", "Mai", "Juin", "Juillet", "Août", "Septembre","Octobre", "Novembre", "Décembre"]
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Module/Global/Views/MonthYearDatePickerView.swift:22]: wont cost that much with localization, `months = [L10n.january, ....]` but i can change to number also np

File: Usiqee/Module/Global/Views/MonthYearDatePickerView.swift [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        fileprivate static let componentsCount = 2
+        fileprivate static let textColor = UIColor.white
+        fileprivate static let textFont = Fonts.MonthYearDatePicker.text
+        fileprivate static let rowHeight : CGFloat = 44
+        fileprivate static let numberOfYears: Int = 5
+    }
+    
+    //MARK: - Properties
+    private var minYear = 2021
+    private var maxYear = 2026
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Module/Global/Views/MonthYearDatePickerView.swift:32]: no, its actually defined by `let numberOfYears: Int = 5` et values are overrided in the inits. this is just the default and worst case values (`Calendar.current` dont return a year 🤷‍♂️)

File: Usiqee/Manager/ManagerAuth.swift [hunk 7]
```diff
@@ -57,6 +57,7 @@ class ManagerAuth {
                 ServiceDeviceToken.shared.unregister(token: token)
             }
         }
+        ManagerEvents.shared.didUpdateFilter()
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Manager/ManagerAuth.swift:60]: reset will erase the selected event types but here is just reload the data

File: Usiqee/Config/FirebaseConfig.swift [hunk 8]
```diff
@@ -13,7 +13,13 @@ import Firebase
    let firebaseEnv = "Release"
 #endif
 
-let db = Firestore.firestore()
+var db: Firestore = {
+    let settings = FirestoreSettings()
+    settings.isPersistenceEnabled = true
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Config/FirebaseConfig.swift:18]: normalement non, je vois pas le rapport mais ça peut etre un ti bug interne du à la persitence. d'ailleurs j'ai zappé de t'en parler, je l'ai laisser pour avoir des data si il ya une coupure d'internet

File: Usiqee/Module/Agenda/EventsFilter/Cells/EventsFilterHeaderCell/EventsFilterHeaderCell.swift [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import UIKit
+
+class EventsFilterHeaderCell: UITableViewHeaderFooterView {
+    
+    //MARK: - Constant
+    enum Constants {
+        static let identifier = "EventsFilterHeaderCell"
+        static let nib = UINib(nibName: identifier, bundle: nil)
+        static let height: CGFloat = 40
+        static let backgroundColor: UIColor = UIColor(hexString: "353537")
```
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Module/Agenda/EventsFilter/Cells/EventsFilterHeaderCell/EventsFilterHeaderCell.swift:17]: we have one mdr

File: Usiqee/Config/FirebaseConfig.swift (hunk 8 above)
[the remark this answers is not in the document]
Comment (med-amine-bj) [reply Usiqee/Config/FirebaseConfig.swift:18]: t'en pense quoi ? je le garde ou on tej ?

[the remark this answers is not in the document]
Comment (guaire94) [reply Usiqee/Config/FirebaseConfig.swift:18]: ouai c parfait comme sa

File: Usiqee/Manager/ManagerAuth.swift (hunk 7 above)
[the remark this answers is not in the document]
Comment (guaire94) [reply Usiqee/Manager/ManagerAuth.swift:60]: okY

File: Usiqee/Module/Global/Views/MonthYearDatePickerView.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (guaire94) [reply Usiqee/Module/Global/Views/MonthYearDatePickerView.swift:22]: as you prefer

File: Usiqee/Module/Agenda/EventsFilter/Cells/EventsFilterEventTypeCell/EventsFilterEventTypeCell.swift [hunk 10]
```diff
@@ -26,7 +26,7 @@ class EventsFilterEventTypeCell: UITableViewCell {
         eventTitle.font = Fonts.EventsFilter.Cell.title
     }
     
-    func configure(event: SelectedEvent) {
+    func configure(event: selectedEventType) {
```
Comment (guaire94) [inline Usiqee/Module/Agenda/EventsFilter/Cells/EventsFilterEventTypeCell/EventsFilterEventTypeCell.swift:29]: selectedEventType -> SelectedEventType

File: Usiqee/Manager/ManagerEvents.swift [hunk 11]
```diff
@@ -7,30 +7,25 @@
 
 import Foundation
 
-typealias SelectedEvent = (event: MEventType, isSelected: Bool)
+typealias selectedEventType = (event: MEventType, isSelected: Bool)
```
Comment (guaire94) [inline Usiqee/Manager/ManagerEvents.swift:10]: selectedEventType -> SelectedEventType

**Review by guaire94 — APPROVED** [review]

**Pull request merged**